tf2_ros/include/tf2_ros/buffer.h

```diff
@@ -65,10 +65,13 @@ namespace tf2_ros
      * @brief  Constructor for a Buffer object
      * @param clock A clock to use for time and sleeping
      * @param cache_time How long to keep a history of transforms
-     * @param debug Whether to advertise the view_frames service that exposes debugging information from the buffer
+     * @param node If passed advertise the view_frames service that exposes debugging information from the buffer
      * @return 
      */
-    TF2_ROS_PUBLIC Buffer(rclcpp::Clock::SharedPtr clock, tf2::Duration cache_time = tf2::Duration(tf2::BUFFER_CORE_DEFAULT_CACHE_TIME));
+    TF2_ROS_PUBLIC Buffer(
+      rclcpp::Clock::SharedPtr clock,
+      tf2::Duration cache_time = tf2::Duration(tf2::BUFFER_CORE_DEFAULT_CACHE_TIME),
+      rclcpp::Node::SharedPtr node = rclcpp::Node::SharedPtr());
```

Comment:
Maybe it adds more complexity than is necessary for the feature, but consider templating on a node-like object so we could pass other types like lifecycle nodes. Similar to other uses of node in tf2_ros, e.g. 

https://github.com/ros2/geometry2/blob/31c537f27dfffc6ebbacb0d03648e73a47ee1e07/tf2_ros/include/tf2_ros/transform_listener.h#L57

Reply:
I would rather keep the scope of this PR limited. That kind of feature enhancement can be done in a follow up PR imo.